tinypico web interface not showing status

### The problem

I just started using tinypico board, which is based on ESP-pico-D4.

When compiling on windows 10 + python 3.9, everything compiles with almost no error
with two exceptions: 
1) hash package is not installed
```
Library Manager: AsyncTCP-esphome @ 1.2.2 is already installed
Library Manager: Installing Hash
Library Manager: Warning! Could not install dependency {'name': 'Hash', 'platforms': ['espressif8266']} for package 'ESPAsyncWebServer-esphome'
Library Manager: Installing ESP8266WiFi
Library Manager: Warning! Could not install dependency {'name': 'ESP8266WiFi', 'platforms': ['espressif8266']} for package 'ESPAsyncWebServer-esphome' 
```
2) a compile error related to HAL-SPI
```
C:\Users\ivan.brunello\.platformio\packages\framework-arduinoespressif32\cores\esp32\esp32-hal-spi.c: In function 'spiTransferBytesNL':
C:\Users\ivan.brunello\.platformio\packages\framework-arduinoespressif32\cores\esp32\esp32-hal-spi.c:922:39: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
                 uint8_t * last_out8 = &result[c_longs-1];
                                       ^
C:\Users\ivan.brunello\.platformio\packages\framework-arduinoespressif32\cores\esp32\esp32-hal-spi.c:923:40: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
                 uint8_t * last_data8 = &last_data;
                                        ^

```

Uploading seems to work correctly

Sensor table is shown but sensor values are blank.


### Which version of ESPHome has the issue?

2022.2.3

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

N/A the local web server is not working

### What platform are you using?

ESP32

### Board

tinypico

### Component causing the issue

web server

### Example YAML snippet

```yaml
esphome:
  name: windows_sensors_hub

esp32:
  board: tinypico
  framework:
      type: arduino
....


web_server:
  port: 80
  auth:
    username: !secret web_user
    password: !secret web_password
....
binary_sensor:
  - platform: status
    name: "windows_sensors_hub_Status"
  - platform: gpio
    name: "window1_chiusura"
    pin: 
      number: GPIO33
    device_class: window
    filters:
    - invert:


....
text_sensor:
  - platform: version
    name: "windows_sensors_hub_esphome_version"
    hide_timestamp: true
  - platform: wifi_info
    ip_address:
      name: "windows_sensors_hub_IP"
      icon: "mdi:ip-network"
    mac_address:
      name: "windows_sensors_hub_mac_address"
      icon: "mdi:ethernet"
```


### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

--- a/esphome/components/json/json_util.cpp
+++ b/esphome/components/json/json_util.cpp
@@ -22,7 +22,7 @@ std::string build_json(const json_build_t &f) {
 #ifdef USE_ESP8266
   const size_t free_heap = ESP.getMaxFreeBlockSize() - 2048;  // NOLINT(readability-static-accessed-through-instance)
 #elif defined(USE_ESP32)
-  const size_t free_heap = heap_caps_get_largest_free_block(MALLOC_CAP_DEFAULT) - 2048;
+  const size_t free_heap = heap_caps_get_largest_free_block(MALLOC_CAP_INTERNAL) - 2048;
 #endif
 
   DynamicJsonDocument json_document(free_heap);
@@ -42,7 +42,7 @@ void parse_json(const std::string &data, const json_parse_t &f) {
 #ifdef USE_ESP8266
   const size_t free_heap = ESP.getMaxFreeBlockSize() - 2048;  // NOLINT(readability-static-accessed-through-instance)
 #elif defined(USE_ESP32)
-  const size_t free_heap = heap_caps_get_largest_free_block(MALLOC_CAP_DEFAULT) - 2048;
+  const size_t free_heap = heap_caps_get_largest_free_block(MALLOC_CAP_INTERNAL) - 2048;
 #endif
 
   DynamicJsonDocument json_document(free_heap);

